app/javascript/controllers/checkbox_controller.js

```diff
@@ -0,0 +1,15 @@
+import { Controller } from '@hotwired/stimulus'
+
+export default class extends Controller {
+  static targets = ['checkbox']
+  static values = { checked: Boolean }
+
+  connect () {
+    this.checked = this.hasCheckboxTarget && this.checkboxTarget.checked
```

Comment:
What's the benefit of maintaining this state? I think you could look at `this.checkboxTarget.checked` inside of `toggle` and not have to worry about maintaining this state.

Reply:
It appears to be unnecessary! Thanks!

Reply:
Might be worth trying to do this with just the stimulus-reveal-controller. Feels like you could put the action directly on the checkbox.

Reply:
@schoork Good catch! I'm pretty sure I can remove `checkbox_controller` since the logic for when to show/hide ended up living outside of it.